server.go

```diff
@@ -255,6 +255,12 @@ type server struct {
 	// agentWhitelist is a list of whitelisted user agent substrings, no
 	// whitelisting will be applied if the list is empty or nil.
 	agentWhitelist []string
+
+	// pendingReconnects is a map of peer addresses that have failed when
+	// connecting outbound.
+	// TODO: Memory leak if outbound target reached and non-persistent peer.
```

Comment:
Let's make this into an issue instead? Also what stops of from periodically garbage collecting this? IIUC this map should never have more than outbound target entries

Reply:
Can also prob just use `sync.Mutex` to not have to worry as much re critical areas.